Makefile.defs

```diff
@@ -12,9 +12,10 @@ endif
 UNAME:=$(shell uname -s)
 UNAME_ARCH:=$(shell uname -m)
 
-# use MAKEFILE_DIR instead of CURDIR for variables that are exported through -include
-MAKEFILE_DIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
-TOOLCHAIN_DIR=$(MAKEFILE_DIR)toolchain
+# Use MAKEFILE_DIR instead of CURDIR for variables that are exported through -include
+# NOTE: Will only behave as long as there aren't any spaces in the paths involved!
+MAKEFILE_DIR:=$(dir $(realpath $(lastword $(MAKEFILE_LIST))))
+TOOLCHAIN_DIR:=$(MAKEFILE_DIR)toolchain
```

Comment:
Why is `$(CURDIR)` not used?

Reply:
Because CURDIR is $PWD at the time make wascalled, which *may* not actually be where the Makefiles are.

(It usually is, though; but it's what I wanted to go with, and we already had it setup, so, eeeh).